lib/ok.ex

```diff
@@ -231,7 +251,21 @@ defmodule OK do
       end
     end
   end
+  defp nest([{:ok, _} = normal | []]) do
+    # last line of the with block is an ok literal
+    quote do
+      unquote(normal)
+    end
+  end
+  
+  defp nest([{{:., _, [{:__aliases__, _, [:OK]}, :success]}, _, _} = normal | []]) do
+    # last line of the with block is an OK.success macro line
+    quote do
+      unquote(normal)
+    end
+  end
```

Comment:
I think that explicitly handling these cases is odd. Someone could write a function that always returned {:ok, value} and it would produce the same errors but not be handled by these clauses. 

I assume that any of the following last lines would also produce warnings that would not be handled by the special case code we have here.

```elixir
# 1
{:error, reason}

# 2
OK.failure(reason)

# 3
_result = {:ok, value}
```

Is there some clever way we can use `Macro.expand` to take the cast ast and expanded it deleting unused paths. It should be possible to make use of the code which generates the warnings.

```elixir
ast = quote do
  case unquote(last_line) do
    {:ok, value} ->
      {:ok, value}
    {:error, reason} ->
      {:error, reason}
  end
end

reduced_ast = expand_or_compile_ast(ast)
```
```